dali/pipeline/pipeline.cc

```diff
@@ -237,14 +238,16 @@ int Pipeline::AddOperator(const OpSpec &spec, const std::string& inst_name) {
 }
 
 int Pipeline::AddOperator(const OpSpec &spec, int logical_id) {
-  return AddOperator(spec, make_string("__", spec.SchemaName(), "_", logical_id), logical_id);
+  std::string name = make_string("__", spec.SchemaName(), "_", logical_id);
+  for (int aux_id = 0; instance_names_.count(name) > 0; aux_id++)
```

Comment:
I don't see when the `instance_names_` is updated. Is it?

And, if we have to keep track of the names explicitly anyway, can we just have some per pipline counter that will generate `aux_id` sequentially (or use instance_names_ size if nothing ever gets removed?) to avoid this loop?

Reply:
There's no way to avoid the loop, because there's no way to stop the user from manually specifying a name that looks like auto-generated.

Reply:
Done.